packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/text/internal/span/CustomStyleSpan.kt

```diff
@@ -59,19 +61,28 @@ public class CustomStyleSpan(
         }
 
   public companion object {
+    private val TAG = CustomStyleSpan::class.simpleName
     private fun apply(
         paint: Paint,
         style: Int,
         weight: Int,
         fontFeatureSettingsParam: String?,
         family: String?,
+        fontVariationSettingsParam: String?,
         assetManager: AssetManager
     ) {
       val typeface =
           ReactTypefaceUtils.applyStyles(paint.typeface, style, weight, family, assetManager)
       paint.apply {
         fontFeatureSettings = fontFeatureSettingsParam
         setTypeface(typeface)
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+          try {
+            fontVariationSettings = fontVariationSettingsParam
+          } catch (e: Exception) {
```

Comment:
```suggestion
          } catch (e: IllegalArgumentExeption) {
```

We only want to be permissive in the case of invalid input, not swallow unexpected other errors. The contract is that this produces [IllegalArgumentException](https://developer.android.com/reference/android/graphics/Paint#setFontVariationSettings(java.lang.String))